server/gql/schema.js

```diff
@@ -19,19 +19,38 @@ const Schema = buildSchema(`
         deleteDataSource(id: Int!): DataSource
         updateDataSource(id: Int!, name: String!, type: DataSourceType!, config: String!): DataSource
         updateSchema(id: Int!, schema: String!): Schema
+        upsertResolver(
+            id: Int
```

Comment:
What's this id?

Reply:
The id of the resolver. If provided it triggers an update, otherwise an insert.

Reply:
Maybe a silly question but doesn't that usually depend on the datasource id?

Reply:
No, Resolver and Data Source are two different things. They are associated (one to many: one data source cab be used by many resolvers) but different tables and GraphQl entities.